lib/ReactViews/Analytics/GeoJsonParameterEditor.tsx

```diff
@@ -18,19 +15,26 @@ import {
 } from "./SelectAPolygonParameterEditor";
 import { getDisplayValue as getRegionPickerDisplayValue } from "./RegionPicker";
 import GeoJsonParameter from "../../Models/FunctionParameters/GeoJsonParameter";
-import { withTranslation } from "react-i18next";
+import { TFunction } from "i18next";
+import { WithTranslation, withTranslation } from "react-i18next";
 import { observer } from "mobx-react";
-import { runInAction } from "mobx";
+import { makeObservable, runInAction } from "mobx";
 import CommonStrata from "../../Models/Definition/CommonStrata";
+import ViewState from "../../ReactViewModels/ViewState";
+
+interface IProps extends WithTranslation {
+  previewed: any;
```

Comment:
I have no idea what the previewed type is supposed to be for these two.

Reply:
Tsc seems to accept that type, thank you.

That uncovered more access to potentially undefined things, so I added some if-statements to ensure things were defined, which made me realize that `selectPolygonOnMap()` in GeoJsonParameterEditor wraps the reading and writing of `this.props.*` inside a `runInAction()`, but RectangleParameterEditor only wraps the writes. I'm guessing `runInAction()` is something related to effects, does the use by RectangleParameterEditor look safe?